crates/agent/src/publications.rs

```diff
@@ -263,31 +263,35 @@ impl Publisher {
 
         // If there are any tests, run them now as long as there's no build errors
         if built.built.built_tests.len() > 0 && !cfg!(test) && built.errors().next().is_none() {
-            let data_plane_job = builds::data_plane(
-                &self.connector_network,
-                &self.bindir,
-                logs_token,
-                &self.logs_tx,
-                tmpdir,
-            );
-            let test_jobs = builds::test_catalog(
-                &self.connector_network,
-                &self.bindir,
-                logs_token,
-                &self.logs_tx,
-                build_id,
-                tmpdir,
-            );
-
-            // Drive the data-plane and test jobs, until test jobs complete.
-            tokio::pin!(test_jobs);
-            let errors: Vec<tables::Error> = tokio::select! {
-                r = data_plane_job => {
-                    tracing::error!(?r, "test data-plane exited unexpectedly");
-                    test_jobs.await // Wait for test jobs to finish.
-                }
-                r = &mut test_jobs => r,
-            }?;
+            tracing::info!(%build_id, %publication_id, tmpdir = %tmpdir.display(), "running tests");
+            let errors = {
```

Comment:
nit: i'm not spotting why this extra nesting is required, or what's actually changed in here

Reply:
Ah, that's due to the fact that we now borrow the built output when running tests, and it conflicted with the mutable borrow when adding test errors to the built output.

I'm now realizing that it may be best to keep test errors as a separate field in `UncommittedBuild`, which would eliminate the need for this nesting and more importantly would allow returning the `TestsFailed` status, which was left as a todo. I'll put up a separate pr for that tomorrow morning.